bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/values/PercentageValue.java

```diff
@@ -69,6 +70,7 @@ public PercentageValue(@Nullable BigDecimal min, @Nullable BigDecimal max, @Null
         }
         this.span = this.max.subtract(this.min);
         this.step = step == null ? BigDecimal.ONE : step;
+        this.stepPercent = this.step.multiply(HUNDRED).divide(this.span, MathContext.DECIMAL128);
```

Comment:
For performance sake: Do we really need `DECIMAL128` or would something lower be sufficient?

Reply:
Well, this is a deeper problem.
The PercentageValue has inputs from 2 sources:
- MQTT will sent values between `min` and `max' or the on/off Strings
- The UI will sent `PercentType`, `IncreaseDecreaseType`, or `UpDownType`
Is that correct?
Then the value will be used in 2 destination:
- MQTT will expect values between `min` and `max`
- The UI will use `PercentType` to render the current value.

The conversion between the MQTT world (min...max) to the percent world (0...100) will have th posibility of loosing precision.
e.g. If the MQTT world is [0...300], then the value incomming value 100 cannot be mapped to the percent value exactly. Then mapping the exact same value back to MQTT will yield a differenet value, because of the rounding needed.

Currently we do not have any information about he presicion, the MQTT world expects.
Do we have any information about our front-end? What precision are ther working with?

Reply:
I was just matching the convention of the file for precision.  I think if the goal is to retain double precision in both percentage and MQTT domain using a 128 bit decimal will work, assuming you round back to doubles at output.  As long as you consistently round before outputting/receiving a value on both MQTT and openhab sides, you shouldn't have any loss of precision.

My MQTT side just expects integers so I'd be fine using floating point values everywhere and rounding.

Reply:
Can one of you help me understand what's needed to submit this PR?  I don't think these issues of precision relate to this specific change but rather to the implementation of the entire class.

Reply:
@jochen314 So you would vote for keeping `DECIMAL128`?

Reply:
At least for now.
I think, it would be another PR change the behaviour of the class in this respect.
And I do not see, what would be the best solution for it